geode-dunit/src/main/java/org/apache/geode/test/dunit/internal/DUnitLauncher.java

```diff
@@ -343,59 +339,120 @@ public static void init(MasterRemote master) {
-      }
+    if (suspectFiles.isEmpty()) {
+      throw new IllegalStateException("No dunit suspect log files found in '"
+          + getDunitSuspectsDir().getAbsolutePath()
+          + "' - perhaps a rule that is cleaning up before suspect processing has already run.");
+    }
+
+    for (File suspect : suspectFiles) {
+      checkSuspectFile(suspect);
+    }
+  }
+
+  public static void checkSuspectFile(File suspectFile) {
+    final List<Pattern> expectedStrings = ExpectedStrings.create("dunit");
+    final LogConsumer logConsumer = new LogConsumer(true, expectedStrings,
+        suspectFile.getName(), 5);
+
+    final StringBuilder suspectStringBuilder = new StringBuilder();
+
+    BufferedReader buffReader;
+    FileChannel fileChannel;
+    try {
+      fileChannel = new FileOutputStream(suspectFile, true).getChannel();
+      buffReader = new BufferedReader(new FileReader(suspectFile));
+    } catch (FileNotFoundException e) {
+      System.err.println("Could not find the suspect string output file: " + e);
+      return;
+    }
+
+    try {
+      String line;
       try {
-        String line;
-        try {
-          while ((line = buffReader.readLine()) != null) {
-            final StringBuilder builder = logConsumer.consume(line);
-            if (builder != null) {
-              suspectStringBuilder.append(builder);
-            }
+        while ((line = buffReader.readLine()) != null) {
+          final StringBuilder builder = logConsumer.consume(line);
+          if (builder != null) {
+            suspectStringBuilder.append(builder);
           }
-        } catch (IOException e) {
-          System.err.println("Could not read the suspect string output file: " + e);
         }
+      } catch (IOException e) {
+        System.err.println("Could not read the suspect string output file: " + e);
+      }
 
-        try {
-          fileChannel.truncate(0);
-        } catch (IOException e) {
-          System.err.println("Could not truncate the suspect string output file: " + e);
-        }
+      try {
+        fileChannel.truncate(0);
+      } catch (IOException e) {
+        System.err.println("Could not truncate the suspect string output file: " + e);
+      }
 
-      } finally {
-        try {
-          buffReader.close();
-          fileChannel.close();
-        } catch (IOException e) {
-          System.err.println("Could not close the suspect string output file: " + e);
-        }
+    } finally {
+      try {
+        buffReader.close();
+        fileChannel.close();
+      } catch (IOException e) {
+        System.err.println("Could not close the suspect string output file: " + e);
       }
+    }
 
-      if (suspectStringBuilder.length() != 0) {
-        System.err.println("Suspicious strings were written to the log during this run.\n"
-            + "Fix the strings or use IgnoredException.addIgnoredException to ignore.\n"
-            + suspectStringBuilder);
+    if (suspectStringBuilder.length() != 0) {
```

Comment:
This will fail on the first log file with suspect strings, and so will not report suspect strings from the remaining log files. If you move this check after the for loop in `closeAndCheckForSuspects()`, it will be able to report suspect strings from all log files.

Reply:
Good point! Functionally it's fixed but I will also apply your other suggestion.